### ISSUE TITLE
"Dropped Frames"?

### ISSUE THREAD

**Issue by ckemere** [Aharoni-Lab/Miniscope-DAQ-QT-Software#14]
Hey!

I just successfully compiled and ran the new software suite with my Miniscope v3 and old acquisition board. I will try to post another issue here in a little bit that lists all the packages that are needed for a clean Ubuntu 20.04 install. (In my 19.10, I had some trouble figuring out which qml-* things I needed.) Anyway, it looks great! I like the extra-long press Record and Stop buttons!

I notice something strange, however. As soon as I start running, my "Dropped Frames" counter starts growing, at maybe ~10 frames per second. But it's **negative** (rather than positive as in the sample image I see on the wiki) - any idea what's going on with that?

thanks!
caleb

**Comment by daharoni** [comment]
Hey,
This is a strange issue. The displayed "Dropped Frames" is calculated by taking (# of frames the DAQ has received from the Miniscope) - (# of frames the Miniscope Software has received over USB). A negative number usually means the DAQ is having issues incrementing its frame counter. It maybe could also mean the DAQ isn't properly sending the frame received value over the UVC control channels.

@ximion , have you experience this?


Also, thanks for considering posting a list of packages for a clean Ubuntu install!

**Comment by ximion** [comment]
@daharoni Yes, this is a known issue, I can definitely reproduce this here.
I haven't looked into this in detail yet, but it's highly likely just like you said: Another case of UVC control channels acting differently on Linux than on Windows (it's actually quite stunning that Windows apparently gives absolutely any value a pass).
In this particular case I would assume range clamping of the value that the `uvcvideo` driver does on Linux according to the UVC spec is to blame, I'll add it to my list of things to look into when time permits.
FWIW, the really dumb reason why this wasn't a high priority for me yet is because my code uses the timestamps the driver assigned to each frame to determine whether there may have been dropped frames, instead of reading the control value (and then the software proceeds to just completely ignore the issue, which is probably not ideal and should at least be reported somewhere :-P)

**Comment by DenisPolygalov** [comment]
I'm experiencing exactly the same issue on Windows 10 machine and your own binary release version: https://github.com/Aharoni-Lab/Miniscope-DAQ-QT-Software/releases/download/v1.02/Miniscope-DAQ-Software_build_v1_02.zip

**Comment by hsw28** [comment]
Any resolution on this? Didn't use to be a problem for me but I can't get anything BUT negative frame rates now. Perhaps related, direct show is also not working... cant remember if it used to!

**Comment by daharoni** [comment]
Very likely this is our software somewhat improperly using a UVC control channel to send number of frames received by the DAQ PCB. Potentially Windows has updated its UVC driver to better follow the UVC guidelines and now our software is failing to send this value correctly.

I'll try to look into this when I have some time.

**Comment by ximion** [comment]
> Very likely this is our software somewhat improperly using a UVC control channel to send number of frames received by the DAQ PCB. Potentially Windows has updated its UVC driver to better follow the UVC guidelines and now our software is failing to send this value correctly.

I think that's highly likely exactly what happened... Switching to different properties for this feature may fix it, but this will require DAQ firmware updates and client tool updates.
